src/main/java/seedu/address/model/item/ItemDate.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.item;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+public class ItemDate {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Dates must follow the format yyyy-mm-dd or dd-mm-yyyy.";
+
+    // TODO: Handle validation
+    public static final String VALIDATION_REGEX = "yyyy-MM-dd";
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(VALIDATION_REGEX);
```

Comment:
I feel that formatter sounds like a function. 
Would this be better as `DATE_TIME_FORMAT` (Static final should be screaming snake case)?

Reply:
Hmm, because it's a formatter object and not really a constant I think. This formatter is passed around as an object and used to validate / format against strings. I'll rename this to something like `DATE_FORMATTER`